components/xnat2bids/Xnat2BidsBidsmap.vue

```diff
@@ -0,0 +1,44 @@
+<template>
+  <fragment>
+    <b-field>
+      <b-switch
+        v-model="value.required"
+        type="is-info"
+        @input="updateValue('required', $event)"
+        >Needs bidsmap file</b-switch
+      >
+    </b-field>
+    <b-field
+      v-if="value.required"
+      label="Directory containing the bidsmap file"
+    >
+      <b-input
+        v-model="value.dir"
+        @input="updateValue('dir', $event)"
+      ></b-input>
+    </b-field>
+    <b-field v-if="value.required" label="Bidsmap file">
+      <b-input
+        v-model="value.file"
+        @input="updateValue('file', $event)"
+      ></b-input>
+    </b-field>
+  </fragment>
+</template>
+
+<script>
+export default {
+  props: {
+    value: {
+      type: Object,
+      required: true,
+    },
+  },
+  methods: {
+    updateValue(key, val) {
+      this.value[key] = val
+      this.$emit('f-bidsmap', this.value)
+    },
```

Comment:
The combination of `v-model` and overwriting the `@input` event and then changing, but also emitting isn't a combination I've seen much (it kind of murkies a one-way data binding and a two-way).  Are the `v-models` really needed? Could those maybe be just `value`s? Or is there maybe an example/documentation that outlines this pattern that could be linked to in a comment?